### ISSUE TITLE
[release-1.25] `etcd.k3s.cattle.io/remove` annotation cannot be used with etcd-only nodes

### ISSUE THREAD

**Issue by brandond** [k3s-io/k3s#6817]
Backport:
* #6816

**Comment by ShylajaDevadiga** [comment]
### Validated on release-1.25 branch using commit id 14f2226b674c1051228cb07e9f503fdb4a4d43e7 
## Environment Details
*Infrastructure*
Cloud EC2 instance

*Node(s) CPU architecture, OS, and Version:*
Ubuntu 20.04

*Cluster Configuration:*
3 etcd only node, 1 cp node

  *Config.yaml:*
```
cat /etc/rancher/k3s/config,yaml
token: <REDACTED>
```
 ## Steps to reproduce the issue  and validate the fix
1. Copy config.yaml
2. Install k3s on 3 nodes as etcd-only mode
```
curl -fL https://get.k3s.io | INSTALL_K3S_VERSION=v1.25.6+k3s1 sh -s - server --cluster-init --disable-apiserver --disable-controller-manager --disable-scheduler

curl -fL https://get.k3s.io | INSTALL_K3S_VERSION=v1.25.6+k3s1 sh -s - server --cluster-init --disable-apiserver --disable-controller-manager --disable-scheduler --server https://<NODE1>:6443
```
3. Install k3s on 4th node as dedicated controlplane node
```
curl -fL https://get.k3s.io | INSTALL_K3S_VERSION=v1.25.6+k3s1 sh -s - server --token secret --disable-etcd --server https://<NODE1>:6443
```
4. Delete one of the etcd only nodes

5. Install etcd-client abd check the member list

**To reproduce:**
```
$ kubectl delete node ip-172-31-2-123 
node "ip-172-31-2-123" deleted
```
etcd node is not removed from the member list after deletion of the node from the cluster
```
$ sudo ETCDCTL_API=3  etcdctl --cert /var/lib/rancher/k3s/server/tls/etcd/client.crt --key /var/lib/rancher/k3s/server/tls/etcd/client.key --cacert /var/lib/rancher/k3s/server/tls/etcd/server-ca.crt member list
8972922a5171713c, started, ip-172-31-2-123-44b575e5, https://172.31.2.123:2380, https://172.31.2.123:2379
aaac80dde6516885, started, ip-172-31-4-136-345ef1b6, https://172.31.4.136:2380, https://172.31.4.136:2379
ae1631713389bf72, started, ip-172-31-7-104-979b3fb6, https://172.31.7.104:2380, https://172.31.7.104:2379
```

**Comment by ShylajaDevadiga** [comment]
Following the same steps, using commit id 14f2226b on release-1.25 branch, found node was not removed from etcd cluster
1. Deleted node from k8s cluster
```
$ kubectl get nodes
NAME               STATUS   ROLES                  AGE   VERSION
ip-172-31-11-204   Ready    etcd                   80s   v1.25.6+k3s-14f2226b
ip-172-31-12-211   Ready    etcd                   44s   v1.25.6+k3s-14f2226b
ip-172-31-12-30    Ready    control-plane,master   86s   v1.25.6+k3s-14f2226b
ip-172-31-4-67     Ready    etcd                   81s   v1.25.6+k3s-14f2226b

$ kubectl delete node ip-172-31-11-204
node "ip-172-31-11-204" deleted

$ kubectl get nodes
NAME               STATUS   ROLES                  AGE    VERSION
ip-172-31-12-211   Ready    etcd                   62s    v1.25.6+k3s-14f2226b
ip-172-31-12-30    Ready    control-plane,master   104s   v1.25.6+k3s-14f2226b
ip-172-31-4-67     Ready    etcd                   99s    v1.25.6+k3s-14f2226b

$ sudo ETCDCTL_API=3  etcdctl --cert /var/lib/rancher/k3s/server/tls/etcd/client.crt --key /var/lib/rancher/k3s/server/tls/etcd/client.key --cacert /var/lib/rancher/k3s/server/tls/etcd/server-ca.crt member list
614f1d7ae6ff052a, started, ip-172-31-4-67-e783f39f, https://172.31.4.67:2380/, https://172.31.4.67:2379/
b0e86b3d397021e9, started, ip-172-31-11-204-9db0600a, https://172.31.11.204:2380/, https://172.31.11.204:2379/
cf455f9a159f37f2, started, ip-172-31-12-211-b9894586, https://172.31.12.211:2380/, https://172.31.12.211:2379/
```
2. Added annotation to the node before deletion, node was not removed from etcd cluster
```
$ kubectl annotate node ip-172-31-11-204 etcd.k3s.cattle.io/remove=true
node/ip-172-31-11-204 annotated

$ kubectl get node ip-172-31-11-204 -o custom-columns=name:.metadata.name,annotations:.metadata.annotations |grep remove
ip-172-31-11-204   map[etcd.k3s.cattle.io/node-address:172.31.11.204 etcd.k3s.cattle.io/node-name:ip-172-31-11-204-358ded88 etcd.k3s.cattle.io/remove:true flannel.alpha.coreos.com/backend-data:{"VNI":1,"VtepMAC":"0a:94:61:9c:c2:84"} flannel.alpha.coreos.com/backend-type:vxlan flannel.alpha.coreos.com/kube-subnet-manager:true flannel.alpha.coreos.com/public-ip:172.31.11.204 k3s.io/hostname:ip-172-31-11-204 k3s.io/internal-ip:172.31.11.204 k3s.io/node-args:["server","--token","********","--disable-apiserver","--disable-controller-manager","--disable-scheduler","--server","https://18.217.253.100:6443/"] k3s.io/node-config-hash:JYKJEWGHKEFGVONCCEEGUR4MWWYUDXX3DMIX3XORA2VQAAD2XVBQ==== k3s.io/node-env:{"K3S_DATA_DIR":"/var/lib/rancher/k3s/data/53e0d162ea659bcfccf8a5abf5e7bf9d36b837a4a9d2f0bd23fae5319a49c7f3"} node.alpha.kubernetes.io/ttl:0 volumes.kubernetes.io/controller-managed-attach-detach:true]

$ sudo ETCDCTL_API=3  etcdctl --cert /var/lib/rancher/k3s/server/tls/etcd/client.crt --key /var/lib/rancher/k3s/server/tls/etcd/client.key --cacert /var/lib/rancher/k3s/server/tls/etcd/server-ca.crt member list
f296ebb0be0559f, started, ip-172-31-12-211-6c9cfce4, https://172.31.12.211:2380/, https://172.31.12.211:2379/
614f1d7ae6ff052a, started, ip-172-31-4-67-8151eada, https://172.31.4.67:2380/, https://172.31.4.67:2379/
ddc6206446f92604, started, ip-172-31-11-204-358ded88, https://172.31.11.204:2380/, https://172.31.11.204:2379/

$ kubectl delete node ip-172-31-11-204
node "ip-172-31-11-204" deleted

$ sudo ETCDCTL_API=3  etcdctl --cert /var/lib/rancher/k3s/server/tls/etcd/client.crt --key /var/lib/rancher/k3s/server/tls/etcd/client.key --cacert /var/lib/rancher/k3s/server/tls/etcd/server-ca.crt member list
f296ebb0be0559f, started, ip-172-31-12-211-6c9cfce4, https://172.31.12.211:2380/, https://172.31.12.211:2379/
614f1d7ae6ff052a, started, ip-172-31-4-67-8151eada, https://172.31.4.67:2380/, https://172.31.4.67:2379/
ddc6206446f92604, started, ip-172-31-11-204-358ded88, https://172.31.11.204:2380/, https://172.31.11.204:2379/
```

**Comment by ShylajaDevadiga** [comment]
### Validated fix using commit id 97100de8d06117ecff5f94eeb7a38e8cbd073fec on release-1.25 branch

**Validated two scenarios**
- Annotate the node to delete
- Delete the node directly from k8s cluster

**1. Annotate the node which deleted the node from etcd cluster and not from k8s cluster**

```
$ sudo ETCDCTL_API=3 etcdctl --cert /var/lib/rancher/k3s/server/tls/etcd/server-client.crt --key /var/lib/rancher/k3s/server/tls/etcd/server-client.key --endpoints https://127.0.0.1:2379 --cacert /var/lib/rancher/k3s/server/tls/etcd/server-ca.crt member list
59f3a97d35419424, started, ip-172-31-14-47-1aeee90b, https://172.31.14.47:2380, https://172.31.14.47:2379
5eea5a67178d8e17, started, ip-172-31-10-156-48cb9b29, https://172.31.10.156:2380, https://172.31.10.156:2379
f719c2f4bd3ac820, started, ip-172-31-12-225-26a04edf, https://172.31.12.225:2380, https://172.31.12.225:2379
```
```
$ kubectl get nodes
NAME               STATUS   ROLES                  AGE     VERSION
ip-172-31-10-156   Ready    etcd                   9m12s   v1.25.6+k3s-97100de8
ip-172-31-10-49    Ready    control-plane,master   9m12s   v1.25.6+k3s-97100de8
ip-172-31-12-225   Ready    etcd                   9m8s    v1.25.6+k3s-97100de8
ip-172-31-14-47    Ready    etcd                   9m12s   v1.25.6+k3s-97100de8
```
```
$ kubectl annotate node ip-172-31-12-225 etcd.k3s.cattle.io/remove=true
node/ip-172-31-12-225 annotated
```
```
$ sudo ETCDCTL_API=3 etcdctl --cert /var/lib/rancher/k3s/server/tls/etcd/server-client.crt --key /var/lib/rancher/k3s/server/tls/etcd/server-client.key --endpoints https://127.0.0.1:2379 --cacert /var/lib/rancher/k3s/server/tls/etcd/server-ca.crt member list
59f3a97d35419424, started, ip-172-31-14-47-1aeee90b, https://172.31.14.47:2380, https://172.31.14.47:2379
5eea5a67178d8e17, started, ip-172-31-10-156-48cb9b29, https://172.31.10.156:2380, https://172.31.10.156:2379
```
```
$ kubectl get nodes
NAME               STATUS   ROLES                  AGE     VERSION
ip-172-31-10-156   Ready    etcd                   9m53s   v1.25.6+k3s-97100de8
ip-172-31-10-49    Ready    control-plane,master   9m53s   v1.25.6+k3s-97100de8
ip-172-31-12-225   Ready    etcd                   9m49s   v1.25.6+k3s-97100de8
ip-172-31-14-47    Ready    etcd                   9m53s   v1.25.6+k3s-97100de8
```
**2. Delete the node from k8s cluster which deletes the node from etcd cluster as well**
```
$ kubectl get nodes
NAME               STATUS   ROLES                  AGE   VERSION
ip-172-31-10-156   Ready    etcd                   11m   v1.25.6+k3s-97100de8
ip-172-31-10-49    Ready    control-plane,master   11m   v1.25.6+k3s-97100de8
ip-172-31-12-225   Ready    etcd                   20s   v1.25.6+k3s-97100de8
ip-172-31-14-47    Ready    etcd                   11m   v1.25.6+k3s-97100de8
```
```
$ sudo ETCDCTL_API=3 etcdctl --cert /var/lib/rancher/k3s/server/tls/etcd/server-client.crt --key /var/lib/rancher/k3s/server/tls/etcd/server-client.key --endpoints https://127.0.0.1:2379 --cacert /var/lib/rancher/k3s/server/tls/etcd/server-ca.crt member list
59f3a97d35419424, started, ip-172-31-14-47-1aeee90b, https://172.31.14.47:2380, https://172.31.14.47:2379
5eea5a67178d8e17, started, ip-172-31-10-156-48cb9b29, https://172.31.10.156:2380, https://172.31.10.156:2379
a79d2f9bbfbd463a, started, ip-172-31-12-225-dc98b2e0, https://172.31.12.225:2380, https://172.31.12.225:2379
```
```
$ kubectl delete node ip-172-31-12-225
node "ip-172-31-12-225" deleted
```
```
$ kubectl get nodes
NAME               STATUS   ROLES                  AGE   VERSION
ip-172-31-10-156   Ready    etcd                   11m   v1.25.6+k3s-97100de8
ip-172-31-10-49    Ready    control-plane,master   11m   v1.25.6+k3s-97100de8
ip-172-31-14-47    Ready    etcd                   11m   v1.25.6+k3s-97100de8
```
```
$ sudo ETCDCTL_API=3 etcdctl --cert /var/lib/rancher/k3s/server/tls/etcd/server-client.crt --key /var/lib/rancher/k3s/server/tls/etcd/server-client.key --endpoints https://127.0.0.1:2379 --cacert /var/lib/rancher/k3s/server/tls/etcd/server-ca.crt member list
59f3a97d35419424, started, ip-172-31-14-47-1aeee90b, https://172.31.14.47:2380, https://172.31.14.47:2379
5eea5a67178d8e17, started, ip-172-31-10-156-48cb9b29, https://172.31.10.156:2380, https://172.31.10.156:2379
```
